sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -117,7 +117,8 @@ class usm_allocator {
 
 private:
   constexpr size_t getAlignment() const {
-    return std::max(alignof(T), Alignment);
+    using std::max; // Cause of MS Windows' macro...
```

Comment:
any quick note how this resolves the build issue? should we move the solution to a common header file?

Reply:
FYI: https://github.com/intel/llvm/commit/e7d4537a73fa40a417ef855b8c12a32a9d9f9839

Reply:
> FYI: [e7d4537](https://github.com/intel/llvm/commit/e7d4537a73fa40a417ef855b8c12a32a9d9f9839)

Hmm, why is that "fix" not enough?

Reply:
There are a few ways to handle this problem.
I suppose the issue can be exposed if someone explicitly include this header and re-define `max` symbol with a macro.
https://github.com/intel/llvm/commit/e7d4537a73fa40a417ef855b8c12a32a9d9f9839 shows one of the approaches to avoid this problem. Codeplay folks applied a bit different solution in the `barrier` extension header - https://github.com/intel/llvm/commit/2e3f61e64e7d9147c40b5b75b0677124ec5ac781.

BTW, I see a few more uses of `std::max` in `sycl/include/CL/sycl/usm.hpp` which probably should be updated as well. I wonder how these are passing `sycl/test/basic_tests/min_max_test.cpp` test. Maybe `-fsycl-device-only` has something to do with it....

Reply:
I think I'll be able to remove the `using` directive here and it would still work. Awaiting local build. Also, I don't need the `std::max`, a `max` would suffice just as well. And I'd expect that we'll have either a macro or `sycl::max` at this point.

Reply:
> I think I'll be able to remove the `using` directive here and it would still work.

This reverts the change. So, how did you manage to get the compilation fail?

Reply:
I'll remove the explicit `std::`, it will be simply `return max(...)`.